### PR TITLE
Windows version updates, travis-ci support

### PR DESCRIPTION
**Pull request by Dudi119**
1) Windows version now supports cppkin as a legit dll, various issues were fixed.
2) example is fixed for windows.
3) Travis script for cppkin.
4) Boost installation will only build needed libs.
5) Thrift installation will only build C++ implementation and tests are switched off.

### CODE REVIEW

File: src/Span.h [hunk 1]
```diff
@@ -31,24 +32,16 @@ namespace cppkin
...
-            return m_timeStamp;
-        }
-        int_fast64_t GetDuration() const{
-            return m_duration;
-        }
-        void SetEndTime(){
-            m_duration = GetCurrentTime() - m_timeStamp;
-        }
+		inline void CreateBinaryAnnotation(const std::string& key, const T& value){}
+		int_fast64_t GetTimeStamp() const;
```
Comment (romakoifman) [inline src/Span.h]: tabs + one not oneliners in .h ?

File: src/cppkin.h [hunk 2]
```diff
@@ -16,8 +16,7 @@
 
 
 #define INIT(params) \
-    core::Environment::Instance().Init(); \
```
Comment (romakoifman) [inline src/cppkin.h]: Remind me why you needed to move this to cpp?

File: src/Span.cpp [hunk 3]
```diff
@@ -30,6 +38,18 @@ namespace cppkin
...
+    int_fast64_t Span::GetDuration() const{
+        return m_duration;
+    }
+
+    void Span::SetEndTime(){
+        m_duration = GetCurrentTime() - m_timeStamp;
+    }
+
     int_fast64_t Span::GetCurrentTime() {
         auto currentTime = chrono::system_clock::now();
```
Comment (romakoifman) [inline src/Span.cpp]: tabs

File: src/cppkin.h (hunk 2 above)
Comment (Dudi119) [reply src/cppkin.h]: Environment contains a global scope external linkage member. If Core will be staticly linked to an App which will also use cppkin, you will have 2 Environments instances. Linux dynamic linker knows how to merge the shared one into the applicative one (done via the GOT section), in windows it needs to be done explictly, Init() function is exported/imported in windows explicitly, which solves the issue.

File: src/Span.h (hunk 1 above)
Comment (Dudi119) [reply src/Span.h]: I will fix this.

File: src/Span.cpp (hunk 3 above)
Comment (Dudi119) [reply src/Span.cpp]: I will fix this.

**Pull request merged**